core/base/src/main/java/org/eclipse/dataspaceconnector/core/security/DefaultPrivateKeyParser.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2022 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.security.KeyParser;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.PrivateKey;
+import java.util.Objects;
+
+/**
+ * Default parser for {@link PrivateKey}.
+ */
+public class DefaultPrivateKeyParser implements KeyParser<PrivateKey> {
+    @Override
+    public boolean canParse(Class<?> keyType) {
+        return Objects.equals(keyType, PrivateKey.class);
+    }
+
+    @Override
+    public PrivateKey parse(String encoded) {
+        var pemParser = new PEMParser(new StringReader(encoded));
+        var converter = new JcaPEMKeyConverter();
+        try {
+            var o = pemParser.readObject();
+            if (o instanceof PEMKeyPair) {
+                var keyPair = (PEMKeyPair) o;
+                return converter.getKeyPair(keyPair).getPrivate();
+            } else if (o instanceof PrivateKeyInfo) {
+                var privateKeyInfo = PrivateKeyInfo.getInstance(o);
+                return converter.getPrivateKey(privateKeyInfo);
+            } else {
+                throw new EdcException("Invalid object type: " + o.getClass());
```

Comment:
Per JavaDoc of `org.bouncycastle.openssl.PEMParser#readObject`, o could be `null` here.

To account for this, it would make sense to add a test case for an empty input file